tests/Unit/MonkeyStubs.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Automattic\LegacyRedirector\Tests\Unit;
+
+use Brain\Monkey;
+use Yoast\WPTestUtils\BrainMonkey\YoastTestCase;
+
+class MonkeyStubs extends YoastTestCase {
+
+	/**
+	 * Sets up test fixtures and additional function stubs.
+	 *
+	 * @return void
+	 */
+	protected function set_up() {
+		parent::set_up();
+
+		Monkey\Functions\stubs(
+			array(
+				'wp_parse_url' => static function ( $url, $component ) {
+					return parse_url( $url, $component );
+				},
+				'esc_url_raw'  => static function( $value ) {
+					return $value;
+				},
```

Comment:
As per https://brain-wp.github.io/BrainMonkey/docs/functions-stubs.html when there is no value/callback provided for the stub, it just returns the first param (as the call to `Monkey\Functions\stubs()` has a `null` 2nd param):

```suggestion
				'esc_url_raw', // Return 1st param unchanged.
```